andy_api/api/intent_processing/move_piece.py

```diff
@@ -130,6 +130,11 @@ def handle(session_id, intent_model, board_str):
 
         # Check if the move is legal
         if check_if_move_legal(board_str, from_location + to_location):
+            # Update stack of board strings with last board string before move
+            board_stack = get_board_stack(session_id)
+            board_stack.push(board_str)
+            set_board_stack(session_id, board_stack)
```

Comment:
Move this to after a move is performed, just in case performing a move errors out.